src/debugConfigProvider.ts

```diff
@@ -181,16 +182,37 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
 
             if (!platformPath) { return null; }
 
-            let debugserverPort = await targetCommand.deviceDebugserver({
-                target: target as Device,
-            });
-            if (!debugserverPort) { return null; }
-
-            dbgConfig.iosDebugserverPort = debugserverPort;
-
             dbgConfig.preRunCommands = (dbgConfig.preRunCommands instanceof Array) ? dbgConfig.preRunCommands : [];
             dbgConfig.preRunCommands.push(`script lldb.target.module[0].SetPlatformFileSpec(lldb.SBFileSpec('${platformPath}'))`);
-            dbgConfig.preRunCommands.push(`process connect connect://127.0.0.1:${debugserverPort}`);
+
+            // iOS 17+ uses Apple LLDB directly to create and attach to the debugserver,
+            // below iOS 17 spawn debugserver with ios-deploy
+            let iosMajorVersion = dbgConfig.iosTarget.version.split('.')[0] as number;
+            if (iosMajorVersion < 17) {
+                let debugserverPort = await targetCommand.deviceDebugserver({
+                    target: target as Device,
+                });
+                if (!debugserverPort) { return null; }
+
+                dbgConfig.iosDebugserverPort = debugserverPort;
+                dbgConfig.preRunCommands.push(`process connect connect://127.0.0.1:${debugserverPort}`);
+            } else {
+                // launch the app via devicectl
+                await _execFile(
+                    'xcrun',
+                    ['devicectl', 'device', 'process', 'launch', '--device', target.udid, '--start-stopped', this.ensureBundleId(dbgConfig)],
+                );
+
+                // attach to the app via LLDB
+                dbgConfig.processCreateCommands = (dbgConfig.postRunCommands instanceof Array) ? dbgConfig.processCreateCommands : [];
+                // LLDB `device` command needs Apple's beta LLDB set as backend for CodeLLDB
+                // e.g. set vscode://settings/lldb.library to /Applications/Xcode-beta.app/Contents/SharedFrameworks/LLDB.framework/Versions/A/LLDB
```

Comment:
I agree that it is an additional hurdle, but I can not think of a easy or clean way to programmatically set this, since we'd be messing with the settings of another app, and the path to Apple's LLDB library might also vary.
Maybe the cleanest option is to just send a clear alert to the user explaining what to do (whenever they are using iOS 17 or when `lldb` complains about `device` not being a command`).